File: src/ros2_bridge/mappers/shape_msgs/plane.rs
//! Typed mapper for `shape_msgs/msg/Plane`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn plane_to_bus(
    msg: ros_env::shape_msgs::msg::Plane,
) -> crate::shape_msgs::msg::v1::Plane {
    crate::shape_msgs::msg::v1::Plane {
        coef: crate::ros2_bridge::mappers::convert::f64_seq(msg.coef),
    }
}

pub(crate) fn plane_to_ros(
    bus: crate::shape_msgs::msg::v1::Plane,
) -> ros_env::shape_msgs::msg::Plane {
    ros_env::shape_msgs::msg::Plane {
        coef: crate::ros2_bridge::mappers::convert::FromF64Seq::from_f64_seq(bus.coef),
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct ShapeMsgsPlaneMapper;

impl TypedTopicMapper for ShapeMsgsPlaneMapper {
    type Ros = ros_env::shape_msgs::msg::Plane;
    type Bus = crate::shape_msgs::msg::v1::Plane;

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(plane_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(plane_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/shape_msgs/plane.py
"""Generated mapper for `shape_msgs/msg/Plane`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.shape_msgs.msg.v1 import Plane as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def plane_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.coef.extend(list(msg.coef))
    return bus


def plane_to_ros(bus):
    from shape_msgs.msg import Plane as RosMsg

    out = RosMsg()
    out.coef = list(bus.coef)
    return out


class ShapeMsgsPlaneMapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from shape_msgs.msg import Plane as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return plane_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return plane_to_ros(bus)
